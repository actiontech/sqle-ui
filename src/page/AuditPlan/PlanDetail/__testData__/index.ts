export const AuditPlanReportList = [
  {
    audit_plan_report_id: 31,
    audit_plan_report_timestamp: '2020-10-19T22:39:57+08:00',
  },
  {
    audit_plan_report_id: 32,
    audit_plan_report_timestamp: '2020-11-19T20:39:57+08:00',
  },
  {
    audit_plan_report_id: 33,
    audit_plan_report_timestamp: '2020-11-19T19:39:57+08:00',
  },
  {
    audit_plan_report_id: 34,
    audit_plan_report_timestamp: '2020-12-19T20:39:57+08:00',
  },
  {
    audit_plan_report_id: 35,
    audit_plan_report_timestamp: '2020-10-19T13:39:57+08:00',
  },
  {
    audit_plan_report_id: 36,
    audit_plan_report_timestamp: '2020-10-19T13:39:57+08:00',
  },
  {
    audit_plan_report_id: 37,
    audit_plan_report_timestamp: '2020-11-19T11:39:57+08:00',
  },
  {
    audit_plan_report_id: 38,
    audit_plan_report_timestamp: '2020-12-19T18:39:57+08:00',
  },
  {
    audit_plan_report_id: 39,
    audit_plan_report_timestamp: '2020-11-19T12:39:57+08:00',
  },
  {
    audit_plan_report_id: 40,
    audit_plan_report_timestamp: '2020-10-19T13:39:57+08:00',
  },
];

export const AuditPlanSqlsRes = {
  head: [
    { name: 'a', desc: '列a', type: 'sql' },
    { name: 'b', desc: '列b' },
  ],
  rows: [
    { a: 'select * from t1 where 1 = 1', b: 'b72' },
    { a: 'select * from t1 where 1 = 2', b: 'b74' },
    { a: 'select * from t1 where 1 = 3', b: 'b76' },
    { a: 'select * from t1 where 1 = 4', b: 'b78' },
    { a: 'select * from t1 where 1 = 5', b: 'b80' },
    { a: 'select * from t1 where 1 = 6', b: 'b82' },
    { a: 'select * from t1 where 1 = 7', b: 'b84' },
    { a: 'select * from t1 where 1 = 8', b: 'b86' },
    { a: 'select * from t1 where 1 = 9', b: 'b88' },
    { a: 'select * from t1 where 1 = 10', b: 'b90' },
  ],
};
export const AuditReport = [
  {
    audit_plan_report_sql: 'select * from t1 where id = t41',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
  {
    audit_plan_report_sql: 'select * from t1 where id = t42',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
  {
    audit_plan_report_sql: 'select * from t1 where id = t43',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
  {
    audit_plan_report_sql: 'select * from t1 where id = t44',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
  {
    audit_plan_report_sql: 'select * from t1 where id = t45',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
  {
    audit_plan_report_sql: 'select * from t1 where id = t46',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
  {
    audit_plan_report_sql: 'select * from t1 where id = t47',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
  {
    audit_plan_report_sql: 'select * from t1 where id = t48',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
  {
    audit_plan_report_sql: 'select * from t1 where id = t49',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
  {
    audit_plan_report_sql: 'select * from t1 where id = t50',
    audit_plan_report_sql_audit_result:
      '[error]禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql\n[notice]不建议使用select *\n[warn]CPU心情不太好\n[normal]所以CPU出门散心去了',
  },
];
