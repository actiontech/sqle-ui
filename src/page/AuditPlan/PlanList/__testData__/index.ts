export const AuditPlanList = [
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'mysql',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app11',
    audit_plan_token:
      'wkpjrnbuneufaljqxijggisjedhpwlhippscklduloxbgymgvwkvhuurmyjiqbaiufvadmkncenueesmbqjqeudkbqdhvihiwzrsdrqbluhashusgzemonxivbsrz',
    audit_plan_meta: {
      audit_plan_params: [],
      audit_plan_type: 'audit_for_java_app',
      audit_plan_type_desc: '审核java应用',
      instance_type: 'mysql',
    },
  },
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'oracle',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app12',
    audit_plan_token:
      'itfpkfctmologkcspmlxwugumdvfslgbgcisqczpehskfjprctfkeuebywxrwsnseltrmsnpwrilahilbxeycpksttfnfbevkrsundytiflrvfccthmrnmorjqfso',
    audit_plan_meta: {
      audit_plan_params: [],
      audit_plan_type: 'audit_for_java_app',
      audit_plan_type_desc: '审核java应用',
      instance_type: 'oracle',
    },
  },
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'mysql',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app13',
    audit_plan_token:
      'aqulzprcpvtydznpplswiijhxsufviotqtfxkocpoukqsiugkmpsnmslnftfdeynbiiobvsguanlktvupflxqaeyjsftedqsuwmxabixdhpekbehnxxqrrjolncmw',
    audit_plan_meta: {
      audit_plan_params: [],
      audit_plan_type: 'audit_for_node_app',
      audit_plan_type_desc: '审核node应用',
      instance_type: 'oracle',
    },
  },
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'oracle',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app14',
    audit_plan_token:
      'nsucohfvciisgsisgythudugjianedqdwvvlefrclxydqxryffrwsunwlhclrboagskhhqpgqthkrwbqhwgziqrdxxswoltlocxaholiwoqpouknqcwelxbekdmjl',
    audit_plan_meta: {
      audit_plan_params: [],
      audit_plan_type: 'slow_log_audit',
      audit_plan_type_desc: '审核慢日志',
      instance_type: 'oracle',
    },
  },
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'mysql',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app15',
    audit_plan_token:
      'oobxrdhaggcwrncxctgtrkmpsrhqlscrxqrrwjqdinxvehvcnhfjmnbefrdmfsuwwfwncqgynnjmwhperpwmmnyhnrszrusrtyaywfuzklnbfvtiaeuxqqhfssuiu',
    audit_plan_meta: {
      audit_plan_params: [],
      audit_plan_type: 'mybatis_audit',
      audit_plan_type_desc: '审核mybatis应用',
      instance_type: 'oracle',
    },
  },
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'oracle',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app16',
    audit_plan_token:
      'ecylwymxtxpvqhkqmtlgqubfudpmurqofvsrvrviihnmzwgtukvqsljvwqdbpxoijeqzvjjttkcmisfspscldihwbktuzuikwxbcstlntldconobxgnqnvoizkunp',
  },
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'mysql',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app17',
    audit_plan_token:
      'rxgloynyjuvfpeghqgkuxcolghxkwmiiebxweiqryuftjpepbshmbqitcloeijaxunwqghlrqytobdrtvkrinsmmpvdvjrhvqnogbbkkrvgxvrmfcjhgsbhjjrvpj',
  },
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'oracle',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app18',
    audit_plan_token:
      'psudtpmsqebdnxjxdtmtripsexqcpvrdhydtnldhslxagcvftnyixhwhurunpqylnrbtgfyxssbjkvfysmojwrvxrqqztutnykjrchjhqasolftsweirknvjrrvlt',
  },
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'mysql',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app19',
    audit_plan_token:
      'tmlgcxbmpcniasdmkqarjereoonwqtohdnfxijmmsuqqkdecclqtbrfrfdnlkxtpwbqlxbzwjwcjfhukjonrjdevqowzcmsvxhgigprwfrkqahwidmjusouiarurg',
  },
  {
    audit_plan_cron: '0 */2 * * *',
    audit_plan_db_type: 'oracle',
    audit_plan_instance_database: 'sqle',
    audit_plan_instance_name: 'db1',
    audit_plan_name: 'audit_for_java_app20',
    audit_plan_token:
      'tidubrvskubhlwtqqexgnjqehlvrgoggmcmwokwetcpmocxyyohgcrctijsmkgiwisetjnursctznwcqayztiwnjnihmphfegfnoubexphwhzzgzjhufculijanuv',
  },
];

export const AuditPlan = {
  audit_plan_cron: '* * * * *',
  audit_plan_db_type: 'oracle',
  audit_plan_instance_database: 'sqle',
  audit_plan_instance_name: 'db1',
  audit_plan_name: 'audit_for_java_app20',
  audit_plan_token:
    'tidubrvskubhlwtqqexgnjqehlvrgoggmcmwokwetcpmocxyyohgcrctijsmkgiwisetjnursctznwcqayztiwnjnihmphfegfnoubexphwhzzgzjhufculijanuv',
  audit_plan_meta: {
    audit_plan_params: [
      { desc: '字段a', key: 'a', type: 'string', value: '123' },
      { desc: '字段b', key: 'b', type: 'int', value: '123' },
      { desc: '字段c', key: 'c', type: 'bool', value: 'true' },
    ],
    audit_plan_type: 'normal',
    audit_plan_type_desc: '普通的SQL审核',
    instance_type: 'oracle',
  },
};
