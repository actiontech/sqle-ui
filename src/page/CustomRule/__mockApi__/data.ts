import { ICustomRuleResV1 } from '../../../api/common';
import { CustomRuleResV1LevelEnum } from '../../../api/common.enum';

export const customRules: ICustomRuleResV1[] = [
  {
    db_type: 'mysql',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.notice,
    rule_id: '1',
    desc: 'name1',
    rule_script: 'w+',
    type: 'rule_type1',
  },
  {
    db_type: 'oracle',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.error,
    rule_id: '2',
    desc: 'name2',
    rule_script: 'w+',
    type: 'rule_type2',
  },
];
