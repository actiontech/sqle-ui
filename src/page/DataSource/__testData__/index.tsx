export const dataSourceList = [
  {
    instance_name: 'db1',
    db_host: '20.20.20.2',
    db_port: '3306',
    db_user: 'root',
    desc: '',
    workflow_template_name: 'default',
  },
];

export const dataSourceInstance = {
  instance_name: 'db1',
  db_host: '20.20.20.2',
  db_port: '3306',
  db_user: 'root',
  db_type: 'mysql',
  desc: '',
  workflow_template_name: 'workflow-template-name-1',
};

export const dataSourceMetas = [
  {
    db_type: 'mysql',
    params: [
      { description: '字段a', name: 'a', type: 'string', value: '123' },
      { description: '字段b', name: 'b', type: 'int', value: '123' },
      { description: '字段c', name: 'c', type: 'bool', value: 'true' },
    ],
  },
  {
    db_type: 'oracle',
    params: [
      { description: '字段e', name: 'e', type: 'string', value: '2222' },
      { description: '字段f', name: 'f', type: 'int', value: '333' },
      { description: '字段mm', name: 'mm', type: 'bool', value: 'false' },
    ],
  },
];
