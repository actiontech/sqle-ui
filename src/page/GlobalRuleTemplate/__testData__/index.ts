export const ruleTemplateListData = [
  {
    db_type: 'mysql',
    desc: '默认规则模板',
    instance_name_list: ['db1'],
    rule_template_name: 'default_mysql',
    id: 1,
  },
  {
    db_type: 'oracle',
    desc: '默认规则模板2',
    instance_name_list: ['db2'],
    rule_template_name: 'default_oracle',
    id: 2,
  },
];

export const ruleTemplateData = {
  rule_template_name: 'default_mysql',
  desc: '默认规则模板',
  db_type: 'mysql',
  instance_name_list: ['db1'],
  rule_list: [
    {
      rule_name: 'all_check_where_is_invalid',
      desc: '禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql',
      value: '',
      level: 'error',
      type: 'DML规范',
      db_type: 'mysql',
    },
    {
      rule_name: 'ddl_check_alter_table_need_merge',
      desc: '存在多条对同一个表的修改语句，建议合并成一个ALTER语句',
      value: '',
      level: 'notice',
      type: '使用建议',
      db_type: 'mysql',
    },
    {
      rule_name: 'ddl_check_column_blob_default_is_not_null',
      desc: 'BLOB 和 TEXT 类型的字段不可指定非 NULL 的默认值',
      value: '',
      level: 'error',
      type: 'DDL规范',
      db_type: 'mysql',
    },
    {
      rule_name: 'ddl_check_column_blob_with_not_null',
      desc: 'BLOB 和 TEXT 类型的字段不建议设置为 NOT NULL',
      value: '',
      level: 'error',
      type: 'DDL规范',
      db_type: 'mysql',
    },
    {
      rule_name: 'ddl_check_column_char_length',
      desc: 'char长度大于20时，必须使用varchar类型',
      value: '',
      level: 'error',
      type: 'DDL规范',
      db_type: 'mysql',
    },
    {
      rule_name: 'ddl_check_column_timestamp_without_default',
      desc: 'timestamp 类型的列必须添加默认值',
      value: '',
      level: 'error',
      type: 'DDL规范',
      db_type: 'mysql',
    },
  ],
  id: 3,
};

export const ruleTemplateDataWithSpecialName = {
  rule_template_name: 'default SQL Server *#&(!#)',
  desc: '默认规则模板',
  db_type: 'mysql',
  instance_name_list: ['db1'],
  rule_list: [
    {
      rule_name: 'all_check_where_is_invalid',
      desc: '禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql',
      value: '',
      level: 'error',
      type: 'DML规范',
      db_type: 'mysql',
    },
    {
      rule_name: 'ddl_check_alter_table_need_merge',
      desc: '存在多条对同一个表的修改语句，建议合并成一个ALTER语句',
      value: '',
      level: 'notice',
      type: '使用建议',
      db_type: 'mysql',
    },
    {
      rule_name: 'ddl_check_column_blob_default_is_not_null',
      desc: 'BLOB 和 TEXT 类型的字段不可指定非 NULL 的默认值',
      value: '',
      level: 'error',
      type: 'DDL规范',
      db_type: 'mysql',
    },
    {
      rule_name: 'ddl_check_column_blob_with_not_null',
      desc: 'BLOB 和 TEXT 类型的字段不建议设置为 NOT NULL',
      value: '',
      level: 'error',
      type: 'DDL规范',
      db_type: 'mysql',
    },
    {
      rule_name: 'ddl_check_column_char_length',
      desc: 'char长度大于20时，必须使用varchar类型',
      value: '',
      level: 'error',
      type: 'DDL规范',
      db_type: 'mysql',
    },
    {
      rule_name: 'ddl_check_column_timestamp_without_default',
      desc: 'timestamp 类型的列必须添加默认值',
      value: '',
      level: 'error',
      type: 'DDL规范',
      db_type: 'mysql',
    },
  ],
  id: 4,
};
