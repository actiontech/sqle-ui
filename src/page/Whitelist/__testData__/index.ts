export const WhitelistData = [
  {
    audit_whitelist_id: 13,
    value: '/* input your sql */\nuse aaa;',
    desc: '123',
    match_type: 'fp_match',
  },
  {
    audit_whitelist_id: 12,
    value: '/* input your sql */123123',
    desc: '',
  },
  {
    audit_whitelist_id: 11,
    value: '\nuse aaa;\n\nuse aaa;',
    desc: '',
    match_type: 'fp_match',
  },
  {
    audit_whitelist_id: 10,
    value: '\nuse aaa;\n\nuse aaa;',
    desc: '',
    match_type: 'fp_match',
  },
  {
    audit_whitelist_id: 9,
    value: '\nuse aaa;\n\nuse aaa;',
    desc: '',
    match_type: 'exact_match',
  },
  {
    audit_whitelist_id: 8,
    value: '\nuse aaa;\n\nuse aaa;',
    desc: '',
  },
  {
    audit_whitelist_id: 7,
    value: '\nuse aaa;\n\nuse aaa;',
    desc: '',
    match_type: 'exact_match',
  },
  {
    audit_whitelist_id: 6,
    value: '\nuse aaa;\n\nuse aaa;',
    desc: '',
    match_type: 'exact_match',
  },
  {
    audit_whitelist_id: 5,
    value: '\nuse aaa;\n\nuse aaa;',
    desc: '',
    match_type: 'exact_match',
  },
  {
    audit_whitelist_id: 4,
    value: '\nuse aaa;\n\nuse aaa;',
    desc: '',
    match_type: 'exact_match',
  },
];
