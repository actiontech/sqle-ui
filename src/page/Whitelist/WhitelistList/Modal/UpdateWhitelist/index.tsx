import { useBoolean } from 'ahooks';
import { Button, Modal } from 'antd';
import { useForm } from 'antd/lib/form/Form';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { useDispatch, useSelector } from 'react-redux';
import auditWhitelist from '../../../../../api/audit_whitelist';
import { IAuditWhitelistResV1 } from '../../../../../api/common.d';
import { ResponseCode } from '../../../../../data/common';
import EmitterKey from '../../../../../data/EmitterKey';
import { ModalName } from '../../../../../data/ModalName';
import { IReduxState } from '../../../../../store';
import { updateWhitelistModalStatus } from '../../../../../store/whitelist';
import EventEmitter from '../../../../../utils/EventEmitter';
import WhitelistForm from '../../../WhitelistForm';
import { WhitelistFormFields } from '../../../WhitelistForm/index.type';

const UpdateWhitelist = () => {
  const { t } = useTranslation();
  const [form] = useForm<WhitelistFormFields>();
  const visible = useSelector<IReduxState, boolean>(
    (state) => !!state.whitelist.modalStatus[ModalName.Update_Whitelist]
  );
  const currentWhitelist = useSelector<
    IReduxState,
    IAuditWhitelistResV1 | null
  >((state) => state.whitelist.selectWhitelist);
  const dispatch = useDispatch();
  const [
    createLoading,
    { setTrue: startCreate, setFalse: createFinish },
  ] = useBoolean();

  const closeModal = React.useCallback(() => {
    dispatch(
      updateWhitelistModalStatus({
        modalName: ModalName.Update_Whitelist,
        status: false,
      })
    );
  }, [dispatch]);

  const submit = React.useCallback(async () => {
    const values = await form.validateFields();
    startCreate();
    auditWhitelist
      .UpdateAuditWhitelistByIdV1({
        audit_whitelist_id: `${currentWhitelist?.audit_whitelist_id}`,
        value: values.sql,
        desc: values.desc,
      })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          EventEmitter.emit(EmitterKey.Refresh_Whitelist_List);
          closeModal();
        }
      })
      .finally(() => {
        createFinish();
      });
  }, [
    closeModal,
    createFinish,
    currentWhitelist?.audit_whitelist_id,
    form,
    startCreate,
  ]);

  React.useEffect(() => {
    if (visible) {
      form.setFieldsValue({
        sql: currentWhitelist?.value,
        desc: currentWhitelist?.desc,
      });
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [visible]);

  return (
    <Modal
      title={t('whitelist.modal.update.title')}
      visible={visible}
      closable={false}
      width={1000}
      footer={
        <>
          <Button onClick={closeModal} disabled={createLoading}>
            {t('common.close')}
          </Button>
          <Button type="primary" onClick={submit} loading={createLoading}>
            {t('common.submit')}
          </Button>
        </>
      }
    >
      <WhitelistForm form={form} />
    </Modal>
  );
};

export default UpdateWhitelist;
