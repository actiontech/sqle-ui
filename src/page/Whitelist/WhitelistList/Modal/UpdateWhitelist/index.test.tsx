import UpdateWhitelist from '.';
import { cleanup, fireEvent, screen, waitFor } from '@testing-library/react';
import {
  mockUseDispatch,
  mockUseSelector,
} from '../../../../../testUtils/mockRedux';
import { SupportTheme } from '../../../../../theme';
import { ModalName } from '../../../../../data/ModalName';
import { renderWithTheme } from '../../../../../testUtils/customRender';
import audit_whitelist from '../../../../../api/audit_whitelist';
import { resolveThreeSecond } from '../../../../../testUtils/mockRequest';
import EventEmitter from '../../../../../utils/EventEmitter';
import EmitterKey from '../../../../../data/EmitterKey';
import { WhitelistData } from '../../../__testData__';
import { UpdateAuditWhitelistReqV1MatchTypeEnum } from '../../../../../api/common.enum';

// https://github.com/react-monaco-editor/react-monaco-editor/issues/176
jest.mock('react-monaco-editor', () => {
  return (props: any) => {
    const { editorDidMount, ...otherProps } = props;
    return <textarea {...otherProps} />;
  };
});

describe.skip('Whitelist/WhitelistList/Modal/UpdateWhitelist', () => {
  let dispatchMock: jest.Mock;

  beforeEach(() => {
    jest.useFakeTimers();
    mockUseSelector({
      user: { theme: SupportTheme.LIGHT },
      whitelist: {
        modalStatus: { [ModalName.Update_Whitelist]: true },
        selectWhitelist: WhitelistData[0],
      },
    });
    const { scopeDispatch } = mockUseDispatch();
    dispatchMock = scopeDispatch;
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  test('should match snapshot', async () => {
    const { baseElement } = renderWithTheme(<UpdateWhitelist />);
    expect(screen.getByLabelText('whitelist.table.desc')).toHaveValue(
      WhitelistData[0].desc
    );
    expect(screen.getByLabelText('order.sqlInfo.sql')).toHaveValue(
      WhitelistData[0].value
    );
    expect(baseElement).toMatchSnapshot();
    cleanup();
    mockUseSelector({
      user: { theme: SupportTheme.LIGHT },
      whitelist: { modalStatus: { [ModalName.Update_Whitelist]: false } },
    });
    const { baseElement: closedContainer } = renderWithTheme(
      <UpdateWhitelist />
    );
    expect(closedContainer).toMatchSnapshot();
  });

  test('should set visible to false when user click closeModal', async () => {
    renderWithTheme(<UpdateWhitelist />);

    fireEvent.click(screen.getByText('common.close'));

    expect(screen.getByLabelText('whitelist.table.desc')).toHaveValue('');
    expect(dispatchMock).toBeCalledTimes(1);
    expect(dispatchMock).toBeCalledWith({
      payload: {
        modalName: 'UPDATE_WHITELIST',
        status: false,
      },
      type: 'whitelist/updateModalStatus',
    });
  });

  test('should send create whitelist request when user click submit button', async () => {
    const createAuditWhitelistSpy = jest.spyOn(
      audit_whitelist,
      'UpdateAuditWhitelistByIdV1'
    );
    createAuditWhitelistSpy.mockImplementation(() => resolveThreeSecond({}));
    const emitSpy = jest.spyOn(EventEmitter, 'emit');

    renderWithTheme(<UpdateWhitelist />);
    fireEvent.input(screen.getByLabelText('whitelist.table.desc'), {
      target: { value: 'whitelist desc' },
    });
    fireEvent.input(screen.getByLabelText('order.sqlInfo.sql'), {
      target: { value: 'select * from table1;' },
    });

    fireEvent.click(screen.getByText('common.submit'));

    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });

    expect(createAuditWhitelistSpy).toBeCalledTimes(1);
    expect(createAuditWhitelistSpy).toBeCalledWith({
      audit_whitelist_id: `${WhitelistData[0].audit_whitelist_id}`,
      desc: 'whitelist desc',
      match_type: UpdateAuditWhitelistReqV1MatchTypeEnum.fp_match,
      value: 'select * from table1;',
    });

    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });

    expect(emitSpy).toBeCalledTimes(1);
    expect(emitSpy).toBeCalledWith(EmitterKey.Refresh_Whitelist_List);
    expect(screen.getByLabelText('whitelist.table.desc')).toHaveValue('');
    emitSpy.mockRestore();
  });
});
