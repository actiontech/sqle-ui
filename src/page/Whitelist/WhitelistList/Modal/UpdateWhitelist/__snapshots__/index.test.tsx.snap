// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Whitelist/WhitelistList/Modal/UpdateWhitelist should match snapshot 1`] = `
<body
  class="ant-scrolling-effect"
  style="width: calc(100% - 0px); overflow: hidden; overflow-x: hidden; overflow-y: hidden;"
>
  <div />
  <div>
    <div
      class="ant-modal-root"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        aria-labelledby="rcDialogTitle0"
        class="ant-modal-wrap"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="ant-modal"
          role="document"
          style="width: 1000px;"
        >
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
          <div
            class="ant-modal-content"
          >
            <div
              class="ant-modal-header"
            >
              <div
                class="ant-modal-title"
                id="rcDialogTitle0"
              >
                whitelist.modal.update.title
              </div>
            </div>
            <div
              class="ant-modal-body"
            >
              <form
                class="ant-form ant-form-horizontal"
              >
                <div
                  class="ant-row ant-form-item ant-form-item-has-success"
                >
                  <div
                    class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                  >
                    <label
                      class=""
                      for="matchType"
                      title="whitelist.table.matchType"
                    >
                      whitelist.table.matchType
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <div
                          class="ant-radio-group ant-radio-group-outline"
                          id="matchType"
                        >
                          <label
                            class="ant-radio-wrapper"
                          >
                            <span
                              class="ant-radio"
                            >
                              <input
                                checked=""
                                class="ant-radio-input"
                                type="radio"
                                value="exact_match"
                              />
                              <span
                                class="ant-radio-inner"
                              />
                            </span>
                            <span>
                              whitelist.matchType.exact
                            </span>
                          </label>
                          <label
                            class="ant-radio-wrapper ant-radio-wrapper-checked"
                          >
                            <span
                              class="ant-radio ant-radio-checked"
                            >
                              <input
                                class="ant-radio-input"
                                type="radio"
                                value="fp_match"
                              />
                              <span
                                class="ant-radio-inner"
                              />
                            </span>
                            <span>
                              whitelist.matchType.fingerPrint
                            </span>
                          </label>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item ant-form-item-has-success"
                >
                  <div
                    class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                  >
                    <label
                      class=""
                      for="desc"
                      title="whitelist.table.desc"
                    >
                      whitelist.table.desc
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <textarea
                          class="ant-input textarea-no-resize"
                          id="desc"
                          placeholder="common.form.placeholder.input"
                          style="height: -40px; min-height: -12px; max-height: -40px; overflow-y: hidden; resize: none; overflow-x: hidden;"
                        >
                          123
                        </textarea>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item ant-form-item-has-success"
                >
                  <div
                    class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                  >
                    <label
                      class="ant-form-item-required"
                      for="sql"
                      title="order.sqlInfo.sql"
                    >
                      order.sqlInfo.sql
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control makeStyles-editor-2 ant-col-xs-24 ant-col-sm-13"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <textarea
                          height="500"
                          id="sql"
                          language="sql"
                          options="[object Object]"
                          theme="vs"
                          width="100%"
                        >
                          /* input your sql */
use aaa;
                        </textarea>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="ant-modal-footer"
            >
              <button
                class="ant-btn"
                type="button"
              >
                <span>
                  common.close
                </span>
              </button>
              <button
                class="ant-btn ant-btn-primary"
                type="button"
              >
                <span>
                  common.submit
                </span>
              </button>
            </div>
          </div>
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
  <textarea
    aria-hidden="true"
    style="letter-spacing:normal;line-height:normal;padding-top:2px;padding-bottom:2px;font-family:-webkit-small-control;font-weight:;font-size:;font-variant:;text-rendering:auto;text-transform:none;width:;text-indent:0;padding-left:2px;padding-right:2px;border-width:1px;box-sizing:;
  min-height:0 !important;
  max-height:none !important;
  height:0 !important;
  visibility:hidden !important;
  overflow:hidden !important;
  position:absolute !important;
  z-index:-1000 !important;
  top:0 !important;
  right:0 !important
"
    tab-index="-1"
  />
</body>
`;

exports[`Whitelist/WhitelistList/Modal/UpdateWhitelist should match snapshot 2`] = `
<body
  class=""
  style=""
>
  <textarea
    aria-hidden="true"
    style="letter-spacing:normal;line-height:normal;padding-top:2px;padding-bottom:2px;font-family:-webkit-small-control;font-weight:;font-size:;font-variant:;text-rendering:auto;text-transform:none;width:;text-indent:0;padding-left:2px;padding-right:2px;border-width:1px;box-sizing:;
  min-height:0 !important;
  max-height:none !important;
  height:0 !important;
  visibility:hidden !important;
  overflow:hidden !important;
  position:absolute !important;
  z-index:-1000 !important;
  top:0 !important;
  right:0 !important
"
    tab-index="-1"
  />
  <div />
</body>
`;
